src/Chapter04.Tests/Listing04.38.Tests.cs

```diff
@@ -1,33 +1,57 @@
 using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
 
 namespace AddisonWesley.Michaelis.EssentialCSharp.Chapter04.Listing04_38.Tests
 {
     [TestClass]
     public class ProgramTests
     {
         [TestMethod]
-        public void ChangeTemperatureWithoutConditionalOperator()
+        public void GivenSegmentsIsNull()
         {
-            bool eventFired = false;
-            Thermostat thermostat = new Thermostat();
-            thermostat.PropertyChanged += (object? sender, System.EventArgs e) =>
-            {
-                eventFired = true;
-            };
-            thermostat.Temperature = 42;
-            Assert.IsTrue(eventFired);
+            const string expected =
+                @"There were no segments to combine.";
+
+            IntelliTect.TestTools.Console.ConsoleAssert.Expect(
+                expected, ()=>Program.Main(null!));
+        }
+
+        [TestMethod]
+        public void GivenSegmentsEmpty()
+        {
+            const string expected =
+                @"There were no segments to combine.";
+
+            IntelliTect.TestTools.Console.ConsoleAssert.Expect(
+                expected, () => Program.Main(Array.Empty<string>()));
+        }
+
+        [TestMethod]
+        public void GivenMultipleSegments()
+        {
+            const string expected =
+                @"Uri: first/second/third";
+
+            IntelliTect.TestTools.Console.ConsoleAssert.Expect(
+                expected, () => Program.Main(new[] { "first", "second", "third" }));
         }
 
-        public static void ChangeHumidityWithConditionalOperator()
+        [TestMethod]
+        public void MyTestMethod()
         {
-            bool eventFired = false;
-            Thermostat thermostat = new Thermostat();
-            thermostat.PropertyChanged += (object? sender, System.EventArgs e) =>
+            string? uri = null;
```

Comment:
You want to avoid strings named things like `uri` as there are built in analyzers that will flag this for not being `System.Uri`